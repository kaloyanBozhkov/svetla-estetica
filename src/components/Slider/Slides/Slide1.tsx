import type { ReactElement } from 'react'
import Button from 'components/UI/Button/Button'
import styles from './slide1.module.scss'

const Slide1 = (): ReactElement => (
    <div className={styles.slide1}>
        <div className={styles.infoArea}>
            <div className={styles.message}>
                <p>Svetla Estetica</p>
                <p>
                    <span>il tuo centro estetico</span>
                    <br />
                    <span>a Dalmine</span>
                </p>
            </div>
            <div className={styles.btnWrapper}>
                <Button
                    label="Listino Prezzi"
                    // eslint-disable-next-line
                    onClick={() => (window.location.hash = 'listino')}
                    modifier="hallow"
                />
            </div>
        </div>
        <img src="/assets/images/sliders/lady.png" alt="lady" />
    </div>
)

export default Slide1
