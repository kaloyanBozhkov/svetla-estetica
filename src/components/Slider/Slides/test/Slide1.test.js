import { shallow } from 'enzyme'
import React from 'react'
import Slide1 from '../Slide1'

describe('testing Slider slides', () => {
    it('should render Slide1', () => {
        expect(shallow(<Slide1 />)).toMatchSnapshot()
    })

    it('should change location href on btn click', () => {
        const wrapper = shallow(<Slide1 />),
            btn = wrapper.find('Button')
        window.location.href = ''
        expect(window.location.href.includes('#listino')).toBeFalsy()
        btn.prop('onClick')()
        expect(window.location.href.includes('#listino')).toBeTruthy()
    })
})
