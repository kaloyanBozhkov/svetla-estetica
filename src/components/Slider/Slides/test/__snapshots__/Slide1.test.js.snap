// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Slider slides should render Slide1 1`] = `
<div
  className="slide1"
>
  <div
    className="infoArea"
  >
    <div
      className="message"
    >
      <p>
        Svetla Estetica
      </p>
      <p>
        <span>
          il tuo centro estetico
        </span>
        <br />
        <span>
          a Dalmine
        </span>
      </p>
    </div>
    <div
      className="btnWrapper"
    >
      <Button
        label="Listino Prezzi"
        modifier="hallow"
        onClick={[Function]}
      />
    </div>
  </div>
  <img
    alt="lady"
    src="/assets/images/sliders/lady.png"
  />
</div>
`;
